lib/cloud_controller/dea/app_stager_task.rb

```diff
@@ -142,15 +165,19 @@ def ensure_staging_is_current!
           raise CloudController::Errors::ApiError.new_from_details('StagingError', "failed to stage application: can't retrieve staging status")
         end
 
-        if @app.staging_task_id != task_id
-          raise CloudController::Errors::ApiError.new_from_details('StagingError', 'failed to stage application: another staging request was initiated')
-        end
+        check_task_id unless use_http
```

Comment:
See above. I think you have made this more complicated when its the same regardless of the path.